qiskit_experiments/calibration/experiments/rabi.py

```diff
@@ -0,0 +1,279 @@
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Rabi amplitude experiment."""
+
+from typing import Any, Dict, List, Optional, Union
+import numpy as np
+
+from qiskit import QiskitError, QuantumCircuit
+from qiskit.circuit import Gate, Parameter
+from qiskit.qobj.utils import MeasLevel
+from qiskit.providers import Backend
+import qiskit.pulse as pulse
+from qiskit.providers.options import Options
+
+from qiskit_experiments.analysis import (
+    CurveAnalysis,
+    CurveAnalysisResult,
+    SeriesDef,
+    fit_function,
+    get_opt_value,
+    get_opt_error,
+)
+from qiskit_experiments.base_experiment import BaseExperiment
+from qiskit_experiments.data_processing.processor_library import get_to_signal_processor
+
+
+class RabiAnalysis(CurveAnalysis):
+    r"""Rabi analysis class based on a fit to a cosine function.
+
+    Analyse a Rabi experiment by fitting it to a cosine function
+
+    .. math::
+        y = a \cos\left(2 \pi {\rm freq} x + {\rm phase}\right) + b
+
+    Fit Parameters
+        - :math:`a`: Amplitude of the oscillation.
+        - :math:`b`: Base line.
+        - :math:`{\rm freq}`: Frequency of the oscillation. This is the fit parameter of interest.
+        - :math:`{\rm phase}`: Phase of the oscillation.
+
+    Initial Guesses
+        - :math:`a`: The maximum y value less the minimum y value.
+        - :math:`b`: The average of the data.
+        - :math:`{\rm freq}`: The frequency with the highest power spectral density.
+        - :math:`{\rm phase}`: Zero.
+
+    Bounds
+        - :math:`a`: [-2, 2] scaled to the maximum signal value.
+        - :math:`b`: [-1, 1] scaled to the maximum signal value.
+        - :math:`{\rm freq}`: [0, inf].
+        - :math:`{\rm phase}`: [-pi, pi].
+    """
+
+    __series__ = [
+        SeriesDef(
+            fit_func=lambda x, a, freq, phase, b: fit_function.cos(
+                x, amp=a, freq=freq, phase=phase, baseline=b
+            ),
+            plot_color="blue",
+        )
+    ]
+
+    @classmethod
+    def _default_options(cls):
+        """Return the default analysis options.
+
+        See :meth:`~qiskit_experiment.analysis.CurveAnalysis._default_options` for
+        descriptions of analysis options.
+        """
+        default_options = super()._default_options()
+        default_options.p0 = {"a": None, "freq": None, "phase": None, "b": None}
+        default_options.bounds = {"a": None, "freq": None, "phase": None, "b": None}
+        default_options.fit_reports = {"freq": "frequency"}
+        default_options.normalization = True
```

Comment:
Can you integrate this into data processor?

Reply:
It already is: the data processor library has an option to add a `MinMaxNormalization` since #93. This seems to be a left over. See here: 6197bfc and here 5bf30ec . Note that the logic in the experiment will change once we have the circuit metadata as you pointed out in #46. We should consider removing `normalization` from the options in `curve_analysis` in a seperate PR.